src/openrct2-ui/windows/Staff.cpp

```diff
@@ -1245,7 +1244,9 @@ void window_staff_overview_tool_down(rct_window* w, rct_widgetindex widgetIndex,
         {
             _staffPatrolAreaPaintValue = PatrolAreaValue::SET;
         }
-        auto staffSetPatrolAreaAction = StaffSetPatrolAreaAction(w->number, destCoords);
+        auto staffSetPatrolAreaAction = StaffSetPatrolAreaAction(
+            w->number, destCoords,
+            _staffPatrolAreaPaintValue == PatrolAreaValue::SET ? StaffSetPatrolAreaMode::Set : StaffSetPatrolAreaMode::Unset);
```

Comment:
Wouldn’t it be cleaner to change the type of `_staffPatrolAreaPaintValue` to `StaffSetPatrolAreaMode`? There seems to be little reason to keep two almost the same enums.

Reply:
No cant really do that without making the window code look a bit rubbish. Window code has 3 states: set, unset, None. Action has 3 states: set, unset, clear. I'd either need to change it so that _staffPatrolAreaPaintValue was a std::optional or I'd have to make it that clear meant none.